matter_server/server/device_controller.py

```diff
@@ -1383,21 +1371,51 @@ async def log_node_long_setup(time_start: float) -> None:
                     # log full stack trace if verbose logging is enabled
                     exc_info=err if LOGGER.isEnabledFor(VERBOSE_LOG_LEVEL) else None,
                 )
-                # NOTE: the node will be picked up by mdns discovery automatically
-                # when it becomes available again.
-                return
+                raise err
 
             # check if this node has any custom clusters that need to be polled
             if polled_attributes := check_polled_attributes(node_data):
                 self._polled_attributes[node_id] = polled_attributes
                 self._schedule_custom_attributes_poller()
-
         finally:
             log_timers[node_id].cancel()
-            self._nodes_in_setup.discard(node_id)
             if is_thread_node:
                 self._thread_node_setup_throttle.release()
 
+    async def _setup_node_with_retry(self, node_id: int) -> None:
+        """Handle set-up of subscriptions and interview (if needed) for known/discovered node."""
+        if node_id not in self._nodes:
+            raise NodeNotExists(f"Node {node_id} does not exist.")
+        if node_id in self._nodes_in_setup:
+            # prevent duplicate setup actions
+            return
+        self._nodes_in_setup.add(node_id)
+
+        node_logger = self.get_node_logger(LOGGER, node_id)
+
+        try:
+            while True:
```

Comment:
instead of this nested try and while loop you could also schedule a timer to retry the task after 60 seconds and abort that timerhandle.

Reply:
Yeah, in the end technically it doesn't matter much. It is work on the asyncio loop which will get executed delayed. I feel the while loop makes it much more readable and easier to reason when retrying actually stops.

Reply:
It doesn't matter - its just a case of preference. Let's leave it this way - if we find it unreadable later we can always make it more pretty. One thing we need to do though is store the reference to the task